documentation/CAMARA-OIDC-profile.md

```diff
@@ -0,0 +1,1022 @@
+<!-- TOC -->
+
+
+## Introduction
+
+Telco network capabilities exposed through APIs provide great value to customers. By simplifying the complexity of telco networks with APIs and making the APIs available across telco networks and countries, CAMARA enables easy and seamless access. To ensure seamless interoperability and increased security, this technical document introduces the OpenID Connect (OIDC) profile, which is specifically tailored for CAMARA APIs.
+
+OpenID Connect is a widely adopted standard for user authentication and authorization in the context of APIs. This document builds on the basic principles of OIDC and outlines CAMARA-specific guidelines to address CAMARA APIs scenarios and use cases.
+
+The CAMARA OIDC profile includes three different authorization mechanisms: Authorization Code Flow, Client Initiated Backchannel Authentication (CIBA), and Client Credentials. Each method is detailed, providing comprehensive insight into the authorization and token request processes, mandatory parameters, successful response content or potential error scenarios.
+
+Key areas covered include:
+
+* Authorization flows:
+
+    * Authorization Code Flow.
+    * Client Initiated Back Channel Authentication (CIBA).
+    * Client Credentials.
+
+* Scope Parameter. Definition of scope parameter format applicable to CAMARA APIs.
+
+* ID Token Content. CAMARA specific details about the content of the Id token.
+
+* Client Authentication. Specifications for client authentication within the CAMARA OIDC profile.
+
+By encapsulating these elements within the CAMARA OIDC profile, this document aims to provide a comprehensive guide for developers and operators, ensuring consistent implementation and adherence to standardized security measures across the CAMARA ecosystem. The defined OIDC profile not only facilitates the integration process, but also serves as a basic framework for developers wishing to leverage the CAMARA APIs while maintaining security and interoperability.
+
+
+## Audience
+
+The target audience for this document is the service/technical departments of operators exposing network functions via standard CAMARA APIs. And applications or client systems that consume CAMARA standard APIs to make use of the operator's network capabilities.
+
+
+## Conventions
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+
+## Authorization using the Authorization Code Flow
+
+The [OAuth 2.0 Authorization Code](https://datatracker.ietf.org/doc/html/rfc6749#section-4.1) flow involves exchanging an authorization code for a token. Instead of requesting authorization directly from the resource owner, the client directs the resource owner to an authorization server (via its user-agent), which in turn directs the resource owner back to the client with the authorization code.
+
```

Comment:
### Cross-Site Request Forgery (CSRF)

Camara choses PKCE for CSRF protection.

Authorization Servers shall require PKCE [RFC7636] with S256 as the code challenge method.

For the Authorization Code flow, Clients shall use PKCE [RFC7636] with S256 as the code challenge method.

Protection provided by state against CSRF is provided by PKCE.
Protection provided by nonce against CSRF is provided by PKCE.